### PR TITLE
fix: type-errors in nextjs

### LINKED ISSUE #3: remove the dynamic proof generation UI & example-circuits
That UI is not needed for the example.

### PR DESCRIPTION
**Pull request by FilipHarald**
closes #3.
depends on #2

### CODE REVIEW

File: packages/nextjs/hooks/noir/useProofGenerator.ts [hunk 1]
```diff
@@ -29,41 +29,34 @@ function getPublicInputsLength(parameters: CircuitAbiParameters) {
...
 }
 
 export const generateProof = async (circuitName: CircuitName, parsedArgs: ParsedArgs) => {
   isGeneratingProof = true;
-  console.log("🧠 start");
-  console.log("parsedArgs", parsedArgs);
   const noir = new NoirBrowser();
   try {
-    const circuit = circuits[circuitName];
+    const circuit = circuits[circuitName] as Circuit<CircuitName>;
```
Comment (urbbru) [inline packages/nextjs/hooks/noir/useProofGenerator.ts:44]: is it needed to cast here?

**Review by urbbru — COMMENTED** [review]
test

File: packages/nextjs/hooks/noir/useProofGenerator.ts (hunk 1 above)
Comment (FilipHarald) [reply packages/nextjs/hooks/noir/useProofGenerator.ts:44]: Well, it was one way to get the type-checking to pass. The types are not set-up perfectly yet, but I want to be able to commit at least. I'm open to another solution!

**Review by ecovirtual — APPROVED** [review]

File: packages/nextjs/hooks/noir/useProofGenerator.ts (hunk 1 above)
Comment (ecovirtual) [reply packages/nextjs/hooks/noir/useProofGenerator.ts:44]: Good to leave a TODO at lest in this PR

File: packages/nextjs/pages/example-zk/GenerateProof.tsx [hunk 2]
```diff
@@ -57,7 +57,11 @@ export const GenerateProof = ({ requiredBirthYear }: { requiredBirthYear: number
       const { proof } = await generateProof("LessThenSignedAge", parsedForm as ParsedArgs);
       setProof(proof);
     } catch (e) {
-      console.error(e.stack);
+      if (e instanceof Error) {
```
[the remark this answers is not in the document]
Comment (FilipHarald) [reply packages/nextjs/pages/example-zk/GenerateProof.tsx:60]: You can `throw` anything in JS. But it's probably going to be an error here anyway. I can do one of below instead.
```
    } catch (e) {
      console.error((e as Error).stack);
    }
```
```
    } catch (e: any) {
      console.error(e.stack);
    }
```

[the remark this answers is not in the document]
Comment (FilipHarald) [reply packages/nextjs/pages/example-zk/GenerateProof.tsx:60]: The second one is what they are using in the rest of the code base.

File: packages/nextjs/hooks/noir/useProofGenerator.ts (hunk 1 above)
Comment (FilipHarald) [reply packages/nextjs/hooks/noir/useProofGenerator.ts:44]: https://github.com/Kryha/scaffold-eth-2-noir/pull/4/commits/cd47494fa9c7fc44e348deb77933e72b53f0faff

**Pull request merged**